File: WebApplication/src/modules/student/models/student-select.dto.ts
import { AutoMap } from "@automapper/classes";
import { NomenclatureDto } from "src/infrastructure/models/nomenclature.dto";
import { EducationType } from "src/modules/application/application-one/enums/education-type.enum";
import { EducationSelectDto } from "src/modules/student/models/education-select.dto";

export class StudentSelectDto {
  @AutoMap()
  studentFullName: string;
  @AutoMap()
  secondNationality: NomenclatureDto;
  @AutoMap()
  uin: string;
  @AutoMap()
  idNumber: string;
  @AutoMap()
  uan: string;
  @AutoMap()
  educationType: EducationType;
  @AutoMap()
  birthDate: Date;
  @AutoMap()
  specialities: EducationSelectDto[] = [];
  @AutoMap()
  doctorals: EducationSelectDto[] = [];
}

File: StudentCredit.Application/Students/Dtos/StudentSelectDto.cs
using StudentCredit.Infrastructure.Helpers.Dtos;

namespace StudentCredit.Application.Students.Dtos
{
    public class StudentSelectDto
    {
        public string StudentFullName { get; set; }
        public NomenclatureDto SecondNationality { get; set; }
        public string Uin { get; set; }
        public string IdNumber { get; set; }
        public string Uan { get; set; }
        public bool HasActiveEducation { get; set; }
        public DateTime? BirthDate { get; set; }
        public List<EducationSelectDto> Specialities { get; set; } = new List<EducationSelectDto>();
        public List<EducationSelectDto> Doctorals { get; set; } = new List<EducationSelectDto>();
	}
}
